scripts/checkUpdates.sh

```diff
@@ -1,49 +1,87 @@
 #!/bin/bash
 
+has_param() {
+    local term="$1"
+    shift
+    for arg; do
+        if [[ $arg == "$term" ]]; then
+            return 0
+        fi
+    done
+    return 1
+}
+
```

Comment:
```suggestion

wait_for_process_to_finish() {
    local process_name="$1"
    while pgrep -a "$process_name" >/dev/null; do 
        sleep 0.1
    done
}

```
Let's implement a function that lets us do a more robust check than a static sleep value.